features/all-positions/LiquidationActionDialog.tsx

```diff
@@ -0,0 +1,488 @@
+  handleClose(): any;
+  isDialogShowing: boolean;
+  selectedSponsor: string | null;
+}
+
+const LiquidationActionDialog = (props: DialogProps) => {
+  const { empState } = EmpState.useContainer();
+  const { dvmState } = DvmState.useContainer();
+  const { finalFee } = dvmState;
+  const { getEtherscanUrl } = Etherscan.useContainer();
+  const { contract: emp } = EmpContract.useContainer();
+
+  const { symbol: tokenSymbol } = Token.useContainer();
+  const {
+    symbol: collSymbol,
+    balance: collBalance,
+    allowance: collAllowance,
+    setMaxAllowance: setMaxCollateralAllowance,
+  } = Collateral.useContainer();
+  const { liquidations } = EmpSponsors.useContainer();
+
+  const [hash, setHash] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const {
+    disputeBondPct,
+    priceIdentifier: priceId,
+    liquidationLiveness,
+    withdrawalLiveness,
+    currentTime,
+  } = empState;
+
+  const prettyBalance = (x: number) => {
+    const x_string = x.toFixed(4);
+    return utils.commify(x_string);
+  };
+
+  const prettyAddress = (x: String) => {
+    return x.substr(0, 6) + "..." + x.substr(x.length - 6, x.length);
+  };
+
+  if (
+    emp &&
+    liquidations &&
+    props.selectedSponsor &&
+    liquidations[props.selectedSponsor] &&
+    liquidationLiveness &&
+    priceId &&
+    disputeBondPct &&
+    collAllowance &&
+    collBalance &&
+    collSymbol &&
+    disputeBondPct &&
+    liquidationLiveness &&
+    withdrawalLiveness &&
+    currentTime &&
+    tokenSymbol &&
+    finalFee
+  ) {
+    const liquidatedPosition = liquidations[props.selectedSponsor];
+    const invertDisputablePrice = isPricefeedInvertedFromTokenSymbol(
+      tokenSymbol
+    );
+    const priceIdUtf8 = utils.parseBytes32String(priceId);
+    const collBalanceNum = collBalance || 0;
+    const finalFeeNum = finalFee || 0;
+    const disputeBondNum = Number(fromWei(disputeBondPct)) || 0;
+
+    const needCollateralAllowance = () => {
+      if (collAllowance === "Infinity") return false;
+      return collAllowance < finalFee;
+    };
+
+    const requiredDisputeBond =
+      Number(fromWei(disputeBondPct)) *
+      Number(liquidatedPosition.lockedCollateral);
+    const collRequired = finalFeeNum + requiredDisputeBond;
+    const collBalanceTooLow = collBalanceNum < collRequired;
+
+    const liquidationTimeRemaining =
+      Number(liquidatedPosition.liquidationTimestamp) +
+      Number(withdrawalLiveness) -
+      Number(currentTime);
+
+    const pendingLiquidationTimeString =
+      liquidationTimeRemaining > 0
+        ? Math.max(0, Math.floor(liquidationTimeRemaining / 3600)) +
+          ":" +
+          Math.max(0, Math.floor((liquidationTimeRemaining % 3600) / 60)) +
+          ":" +
+          Math.max(0, (liquidationTimeRemaining % 3600) % 60)
+        : "None";
+
+    const translateLiquidationStatus = (
```

Comment:
@nicholaspai don't we do this same sort of conversion elsewhere (I thought I saw it in a PR you wrote recently)? Thoughts on making this logic common, especially since it's a little tricky?

Reply:
something [similar](https://github.com/UMAprotocol/emp-tools/blob/615047591c37a98e27817d784ac498eaa6de71bc/features/all-positions/AllLiquidations.tsx#L145) is in `AllLiquidations` but it does not contain all the extra info that this one has in the `description` field.

I think for now it's fine to have it somewhat repetitive and then we can pull this logic out into a separate utility when we solve #64

Reply:
SGTM

Reply:
Yeah I had the same thought but the copy is different enough that I thought no need to overoptimize